dump/parser.go

```diff
@@ -54,6 +65,43 @@ func Parse(r io.Reader, h ParseHandler, parseBinlogPos bool) error {
 			return c == '\r' || c == '\n'
 		})
 
+		// parsed gtid set from mysqldump, refer to canal_test.go TestDumperHandler function
+		if parseBinlogPos && !gtidDoneParsed && !binlogParsed {
+			if m := oneGtidExp.FindAllStringSubmatch(line, -1); len(m) == 1 {
+				gset := m[0][1]
+				if err := h.UpdateGtidFromPurged(gset); err != nil {
+					return errors.Trace(err)
+				}
+				gtidDoneParsed = true
+			}
+			if m := mutilGtidStartExp.FindAllStringSubmatch(line, -1); len(m) == 1 {
+				gset := m[0][1]
+				if err := h.UpdateGtidFromPurged(gset); err != nil {
+					return errors.Trace(err)
+				}
+				mutilGtidParsed = true
+			}
+
+			if mutilGtidParsed && !gtidDoneParsed {
+				if m := midUuidSet.FindAllStringSubmatch(line, -1); len(m) == 1 {
+					gset := m[0][1]
+					if err := h.UpdateGtidFromPurged(gset); err != nil {
+						return errors.Trace(err)
+					}
+
+				}
+
+				if m := endUuidSet.FindAllStringSubmatch(line, -1); len(m) == 1 {
+					gset := m[0][1]
+					if err := h.UpdateGtidFromPurged(gset); err != nil {
+						return errors.Trace(err)
+					}
+					gtidDoneParsed = true
+				}
+
+			}
+		}
```

Comment:
these codes want to get all gtid set in dump file, and then update dumpParseHandler's `gset`?
can we make the code more simple?